src/Core__Object.res

```diff
@@ -1,31 +1,250 @@
+
+## Examples
+
+```rescript
+let fruit = Symbol.make("fruit")
+let x = Object.empty()
+x->Object.setSymbol(fruit, "banana")
+x->Object.getSymbol(fruit) // Some("banana")
+```
+*/
+@get_index
+external getSymbol: ({..}, Core__Symbol.t) => option<'a> = ""
 @get_index external getSymbolUnsafe: ({..}, Core__Symbol.t) => 'a = ""
 
-@set_index external set: ({..}, string, 'a) => unit = ""
+/**
+`set(name, value)` assigns a value to the named object property, overwriting the previous value if any. See [Working with Objects on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#objects_and_properties)
+
+## Examples
+
+```rescript
+{"a": 1}->Object.set("a", 2) // {"a": 2}
+{"a": 1}->Object.set("a", None) // {"a": None}
+{"a": 1}->Object.set("b", 2) // {"a": 1, "b": 2}
+```
+*/
+@set_index
+external set: ({..}, string, 'a) => unit = ""
+
 @set_index external setSymbol: ({..}, Core__Symbol.t, 'a) => unit = ""
 
-@val external keysToArray: {..} => array<string> = "Object.keys"
+/**
+`keysToArray` returns an array of an object's own enumerable string-keyed property names. See [ECMAScript Language Specification](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-object.keys) 
+or [Object.keys on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys).
+
+## Examples
+
+```rescript
+{"a": 1, "b": 2}->Object.keysToArray // ["a", "b"]
+{"a": None}->Object.keysToArray // ["a"]
+Object.empty()->Object.keysToArray // []
+```
+*/
+@val
+external keysToArray: {..} => array<string> = "Object.keys"
 
 @val external hasOwnProperty: ({..}, string) => bool = "Object.prototype.hasOwnProperty.call"
 
-@val external seal: 'a => 'a = "Object.seal"
-@val external preventExtensions: 'a => 'a = "Object.preventExtensions"
-@val external freeze: 'a => 'a = "Object.freeze"
+/**
+`seal` seals an object. Sealing an object prevents extensions and makes existing properties non-configurable. A sealed object has a fixed set of properties. Unlike `freeze`, values of existing properties can still be changed as long as they are writable. 
+
+**Note:** `seal` returns the same object that was passed in; it does not create a copy. Any attempt to delete or add properties to a sealed object will fail, either silently or by throwing an error. 
+
+See [ECMAScript Language Specification](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-object.seal) and [Object.seal on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/seal)
+
+## Examples
+
+```rescript
+let point = {"x": 1, "y": 2}
+point->Object.set("x", -7) // succeeds
+point->Object.seal->ignore
+point->Object.set("z", 9) // fails
+point->Object.set("x", 13) // succeeds
+```
+*/
+@val
+external seal: 'a => 'a = "Object.seal"
```

Comment:
Should be:

```suggestion
external seal: ({..} as 'a) => 'a = "Object.seal"
```

Same with the other additions that uses type variables without the object constraint.

Reply:
Wow didn't know about this `as` keyword. Its not on the ReScript web site syntax lookup.

Reply:
There is unfortunately quite a few things missing in the language manual. Mostly advanced concepts that you will rarely come across though, unless you meet someone familiar with OCaml or ReasonML.